File: totaleconomy-api/src/main/java/com/ericgrandt/totaleconomy/command/CommandArgument.java
package com.ericgrandt.totaleconomy.command;

/**
 * Represents a parsed command argument.
 * <p>
 * This sealed hierarchy ensures only defined argument types ({@link DoubleArg}, {@link StringArg}, {@link PlayerArg},
 * etc.) can be used.
 * </p>
 */
public sealed interface CommandArgument permits DoubleArg, StringArg, PlayerArg {
}

File: totaleconomy-api/src/main/kotlin/com/ericgrandt/totaleconomy/command/CommandArgument.kt
package com.ericgrandt.totaleconomy.command

import com.ericgrandt.totaleconomy.model.Player

sealed class CommandArgument {
    data class DoubleParam(
        val value: Double,
    ) : CommandArgument()

    data class PlayerParam(
        val value: Player,
    ) : CommandArgument()

    data class StringParam(
        val value: String,
    ) : CommandArgument()
}
